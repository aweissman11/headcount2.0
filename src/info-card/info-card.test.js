import React from 'react';
import ReactDom from 'react-dom';
import { shallow, mount } from 'enzyme';
import kinderData from '../data/kindergartners_in_full_day_program.js';
import DistrictRepository from '../helper';

import InfoCard from './info-card';

describe('info-card', () => {
  let data;
  let compareMock;
  let wrapper;

  beforeEach(() => {
    data = {"stats":{"2004":0.24,"2005":0.278,"2006":0.337,"2007":0.395,"2008":0.536,"2009":0.598,"2010":0.64,"2011":0.672,"2012":0.695,"2013":0.703,"2014":0.741},"location":"COLORADO"}
    compareMock = jest.fn()
    wrapper = shallow(<InfoCard data={data} compareData={compareMock}/>)
  })

  it('should render the correct elements', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should compare data', () => {
    wrapper.simulate('click')
    expect(compareMock).toHaveBeenCalled()
  })


})