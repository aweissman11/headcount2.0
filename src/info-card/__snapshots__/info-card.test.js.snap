// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`info-card should render the correct elements 1`] = `
<div
  className="info-card"
  onClick={[Function]}
>
  <h4>
    COLORADO
  </h4>
  <div
    className="year-list"
  >
    <p
      className="low"
      key="0.242004"
    >
      2004
      : 
      0.24
      <span
        className="emoji"
      >
        ...😡
      </span>
    </p>
    <p
      className="low"
      key="0.2782005"
    >
      2005
      : 
      0.278
      <span
        className="emoji"
      >
        ...😡
      </span>
    </p>
    <p
      className="low"
      key="0.3372006"
    >
      2006
      : 
      0.337
      <span
        className="emoji"
      >
        ...😡
      </span>
    </p>
    <p
      className="low"
      key="0.3952007"
    >
      2007
      : 
      0.395
      <span
        className="emoji"
      >
        ...😡
      </span>
    </p>
    <p
      className="hi"
      key="0.5362008"
    >
      2008
      : 
      0.536
      <span
        className="emoji"
      >
        ...👍
      </span>
    </p>
    <p
      className="hi"
      key="0.5982009"
    >
      2009
      : 
      0.598
      <span
        className="emoji"
      >
        ...👍
      </span>
    </p>
    <p
      className="hi"
      key="0.642010"
    >
      2010
      : 
      0.64
      <span
        className="emoji"
      >
        ...👍
      </span>
    </p>
    <p
      className="hi"
      key="0.6722011"
    >
      2011
      : 
      0.672
      <span
        className="emoji"
      >
        ...👍
      </span>
    </p>
    <p
      className="hi"
      key="0.6952012"
    >
      2012
      : 
      0.695
      <span
        className="emoji"
      >
        ...👍
      </span>
    </p>
    <p
      className="hi"
      key="0.7032013"
    >
      2013
      : 
      0.703
      <span
        className="emoji"
      >
        ...👍
      </span>
    </p>
    <p
      className="hi"
      key="0.7412014"
    >
      2014
      : 
      0.741
      <span
        className="emoji"
      >
        ...👍
      </span>
    </p>
  </div>
</div>
`;
